src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -327,9 +334,15 @@ pub trait TimestampProvider {
         }
 
         // If we've acquired a read timestamp from the timestamp oracle, use it
-        // as the new lower bound for the candidate
+        // as the new lower bound for the candidate.
+        // In Strong Session Serializable, we ignore the oracle timestamp for now, unless we need
+        // to use it.
         if let Some(timestamp) = &oracle_read_ts {
-            candidate.join_assign(timestamp);
+            if isolation_level != &IsolationLevel::StrongSessionSerializable
+                || when.must_advance_to_timeline_ts()
```

Comment:
What happens without this?

Reply:
Everything is still correct without this, but it's unnecessarily aggressive in moving the timestamp forward. If we always select global timestamp oracle's read timestamp in Strong Session Serializable, then it ends up having identical behavior to Strict Serializable.

Reply:
I meant without the `when.must_advance_to_timeline_ts()` part. Is that needed because before this always applied, and in some cases even with SSS it still needs to?

Also why are we doing this for Serializable? Shouldn't it be able to not advance to the oracle ts?

Reply:
Oh gotcha, all isolation levels (including Serializable) need to advance to the timestamp oracle for read-then-write plans. This is for scenarios like the following:

1. Table `t` exists, since: `1`, upper: `2`.
2. Index `i` exists on `t`, since: `1`, upper: `2`.
3. Timestamp oracle read timestamp is `1`.
4. Write to table `t`. 
5. Oracle read timestamp advances to `2`.
6. Table upper advances to `2`.
7. Index does not advance yet.
8. Update to table `t`.

The read portion of the update will use the index (assuming we are on the right cluster). It's not enough to read at the upper of the index minus one, `1`. We must read at `2` so we can observe the most recent write. This gets even worse if we replace step 4 with an update/delete instead of a write. Then we end up with invalid retractions and the whole table becomes unreadable.

Previously we were implicitly relying on `oracle_read_ts` to be `Some` if `when.must_advance_to_timeline_ts()`. Technically we still are relying on that. We probably want to stop and return an error if the following is ever true `when.must_advance_to_timeline_ts() && oracle_read_ts.is_none()`, otherwise a bug may lead to a corrupt table.